lib/if.h

```diff
@@ -12,6 +12,10 @@
 #include "qobj.h"
 #include "hook.h"
 #include "admin_group.h"
+#ifdef GNU_LINUX
```

Comment:
IMO, let's remove the zebra.h inclusion of net/if.h and move it into this file instead.  It will require some small reworking of stuff in nhrpd/linux.c and zebra/if_netlink.c, but frankly it's worth it.

I'd also break up the commits into 2 -> The inclusion of the new header files and the small rearrangement and the second commit would be the actual changes in lib/if.c

Why do I want it this way?  too much is in zebra.h -> It needs to be broken up.  if.h should own interface header definitions not zebra.h.  I've started breaking up zebra.h in other commits just haven't gotten to this part yet.  Since you are in here I would just do it now.

Reply:
Not sure we can remove "net/if.h", it is even set in Linux kernel sources #include.

> I'd also break up the commits into 2 -> The inclusion of the new header files and the small rearrangement and the second commit would be the actual changes in lib/if.c

done

In lib/if.c, I am testing the definition IFF_LOWER_UP

Reply:
Please reread what I stated.  inclusion of net/if.h does not belong in zebra.h it belongs in lib/if.h.